File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기(삽입정렬).py
import sys
N = int(sys.stdin.readline())
arr=[]
for i in range(N):
    arr.append(int(sys.stdin.readline()))

def insertion_sort(A):
    n = len(A)
    for i in range(1,n):
        key = A[i]
        j = i-1
        while j>=0 and A[j]>key:
            A[j+1] = A[j]
            j-=1
        A[j+1] = key

insertion_sort(arr)
for i in arr:
    print(i)

File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기(삽입정렬).js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const N = Number(input[0]);
const arr = [];
for(let i=1; i<=N;i++){
    arr.push(Number(input[i]));
}

function insertion_sort(A){
    const n = A.length;
    for(let i=1; i<n; i++){
       let key = A[i];
       j = i-1;
       while(j>=0 && A[j]>key){
        A[j+1]=A[j];
        j--;
       }
       A[j+1] = key;
    }
}


insertion_sort(arr);

console.log(arr.join(""));
